src/components/RoomDropdown/index.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useEffect, useRef } from 'react'
+import { useState } from 'react'
+import styled from 'styled-components'
+import { KARELIA } from '../../utils/css'
+
+type Props = {
+    button?: any
+}
+
+const DropDownContainer = styled.div`
+    margin-left: 10px;
+`
+const DropDownHeader = styled.div`
+    margin-bottom: 0.8em;
+    padding: 0 2em 0 1em;
+    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.15);
+    font-weight: 500;
+    font-size: 1.3rem;
+    color: #3faffa;
+    background: #ffffff;
+`
+const DropDownListContainer = styled.div`
+    position: absolute;
+    top: 75px;
+    right: 35px;
+    z-index: 100;
+`
+const DropDownList = styled.table`
+    width: 250px;
+    padding: 0;
+    margin: 0;
+    background: #ffffff;
+    border-radius: 10px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
+    box-sizing: border-box;
+    color: #59799c;
+    font-family: ${KARELIA};
+    font-size: 14px;
+    list-style-type: none;
+    border-spacing: 0px;
+    hr {
+        border: 0;
+        border-top: 1px solid #dee6ee;
+        padding: 0px;
+        margin: 0px;
+    }
+`
+const ListItem = styled.tr`
+    padding: 10px 15px;
+    background-clip: padding-box;
+    display: flex;
+    td {
+        line-height: 13px;
+        &:last-child {
+            margin-left: 15px;
+        }
+    }
+    &:first-child {
```

Comment:
Pseudo-selectors by default hook up to the styled component they style.
```css
:first-child {}
:hover {}
/* the above is the same as the following */
&:first-child {}
&:hover {}
```

That's my impression. Correct me if I'm wrong.

The above does not work for attribute-based selectors, e.g.

```css
&[disabled] {}
/* Is different than */
[disabled] {}
```

Cheers.

Reply:
Yup this is correct! I've changed it